src/OrchardCore.Modules/OrchardCore.Flows/Indexing/BagPartIndexHandler.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using OrchardCore.Flows.Models;
+using OrchardCore.Indexing;
+
+namespace OrchardCore.Flows.Indexing
+{
+    public class BagPartIndexHandler : ContentPartIndexHandler<BagPart>
+    {
+        private readonly IServiceProvider _serviceProvider;
+
+        public BagPartIndexHandler(IServiceProvider serviceProvider)
+        {
+            _serviceProvider = serviceProvider;
+        }
+
+        public override async Task BuildIndexAsync(BagPart bagPart, BuildPartIndexContext context)
+        {
+            var options = context.Settings.ToOptions();
+            if (options == DocumentIndexOptions.None)
+            {
+                return;
+            }
+
+            if (bagPart.ContentItems.Count != 0)
+            {
+                // Lazy resolution to prevent cyclic dependency
+                var contentItemIndexHandlers = _serviceProvider.GetServices<IContentItemIndexHandler>();
+
+                foreach (var contentItemIndexHandler in contentItemIndexHandlers)
+                {
+                    foreach (var contentItem in bagPart.ContentItems)
+                    {
+                        var buildIndexContext = new BuildIndexContext(context.DocumentIndex, contentItem, contentItem.ContentType);
```

Comment:
@sebastienros asked: When a bag part content item is indexed, shouldn't the keys be prefixed by the container keys?

Reply:
Good point. I will add the prefix using $"{context.Key}.{contentItem.ContentType}".

Reply:
BagParts can be named, so i think it should use the part name. Please try the case when two BapParts with difference names are added to ensure it works fine -> we get the fields indexes under different prefixes.

Reply:
Let me try that.

Reply:
What about the aspects and default fields of a content item (AspectsContentIndexHandler and DefaultContentIndexHandler) ... currently they are all keyed off the values in IndexingConstants. Shouldn't these respect the context's key as well?

Reply:
ContentPartIndexHandler<TPart> doesn't honor this either. It uses the part definition name as the key with no prefix. Neither does ContentPickerFieldIndexHandler - is uses the context's key as-is. ContentFieldIndexHandler<TField> handles this only partially - it just uses the part definition name as the prefix followed by the field definition name.